internal/yaml/yaml.go

```diff
@@ -4,13 +4,63 @@ import (
 	"bufio"
 	"bytes"
 	"os"
+	"path"
+	"path/filepath"
 	"strconv"
 	"strings"
 
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v3"
 )
 
+// Read reads all files in the given filenames having a .yaml or .yml extension,
+// joins them with "\n---\n" and returns the result. For any filename that is a
+// directory, it recursively reads all files in the directory. This function
+// performs no validation on the contents of the files.
+func Read(filenames []string) ([]byte, error) {
+	var allBytes [][]byte
+	for _, filename := range filenames {
+		var err error
+		if filename, err = filepath.Abs(filename); err != nil {
+			return nil, err
+		}
+		fileInfo, err := os.Stat(filename)
+		if err != nil {
+			return nil, err
+		}
+		var fileBytes []byte
+		if fileInfo.IsDir() {
+			var dirEntries []os.DirEntry
+			if dirEntries, err = os.ReadDir(filename); err != nil {
+				return nil, err
+			}
+			dirFiles := make([]string, 0, len(dirEntries))
+			for _, dirEntry := range dirEntries {
+				dirFiles = append(dirFiles, path.Join(filename, dirEntry.Name()))
+			}
+			if fileBytes, err = Read(dirFiles); err != nil {
```

Comment:
This infinite recursion makes me slightly uncomfortable, especially because we use `os.Stat` (which follow symlinks) while _also_ reading files endlessly without some limit.

Because of this, I think someone could create a file, and then a symlink to "jump back" to the root of the directory, causing the memory to endlessly fill up.

To guard against this, I would at least use a `bytes.Buffer` with an upper limit or maybe even ignore symlinks and/or prevent infinite recursion by other means.

Reply:
I overlooked the possibility of symlinks. 🤦‍♂️

Reply:
I fixed this. We follow symlinks, but ignore any symlink that evaluates to a path that contains itself.

There is still no explicit limit on how much we'll read, but reading infinitely is no longer a possibility.